server/internal/core/file/file_test.go

```diff
@@ -0,0 +1,54 @@
+package file
+
+import (
+	"testing"
+
+	"github.com/google/uuid"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/oxidrive/oxidrive/server/internal/core/user"
+	"github.com/oxidrive/oxidrive/server/internal/testutil"
+)
+
+func Test_Create(t *testing.T) {
+	t.Run("creates a new valid file", func(t *testing.T) {
+		t.Parallel()
+
+		file, err := Create(nil, "this/is/a/directory/filename.txt", 5, user.ID(testutil.Must(uuid.NewV7())))
+
+		assert.NotNil(t, file)
+		assert.NoError(t, err)
+	})
+
+	t.Run("returns an error with an invalid path", func(t *testing.T) {
+		t.Parallel()
+
+		file, err := Create(nil, "/this/is/a/directory/filename.txt", 5, user.ID(testutil.Must(uuid.NewV7())))
+
+		assert.Nil(t, file)
+		assert.ErrorIs(t, err, ErrInvalidPath)
+	})
+}
+
+func TestFile_isValid(t *testing.T) {
```

Comment:
Testing a private function is not recommended, what we need to ensure is working is the public API (aka public methods in this case).

I think these test cases can be simply put in the `Create` suite, checking `assert.ErrorIs(t, err, testCase.expected)`